src/lib/share/__tests__/generateShareLink.js

```diff
@@ -89,9 +89,9 @@ describe('generateShareLink', () => {
     const params = { key: 'value with spaces' }
 
     // When
-    const link = generateShareLink(action, params)
+    const link = encodeURI(generateShareLink(action, params))
```

Comment:
The link that we generate should be already encoded.
This suggest there are probably issues at the code where we use links that are not encoded.
I suggest we change the generateShareLink to encode the uri and remove encodeuri from other places

Reply:
The generateShareObject method does encodeURI that breaks the reference in Share Object . But most likely it is necessary to remove reuse in the generateShareObject method

Reply:
[done](https://github.com/GoodDollar/GoodDAPP/pull/615/commits/df8a4ab183a99efc9832f9eeba4ad3c9830006ec)